pkg/api/serialization_test.go

```diff
@@ -217,6 +217,9 @@ func fuzzInternalObject(t *testing.T, forVersion string, item runtime.Object, se
 				Kind: "Service",
 				Name: j.To.Name,
 			}
+			if j.TLS != nil && len(j.TLS.Termination) == 0 && len(j.TLS.DestinationCACertificate) == 0 {
```

Comment:
I would think this would work as well:
```
func(j *route.TLSConfig, c fuzz.Continue) {
	c.FuzzNoCustom(j)
	if j != nil && len(j.Termination) == 0 && len(j.DestinationCACertificate) == 0 {
		j.Termination = route.TLSTerminationEdge
	}
},
```

SInce the conversion works at the TLSConfig object level, I think fuzzing should as well. Otherwise, if we started using TLSConfig in another object, our fuzzing would be out of date. I don't think you need the change in the RouteSpec fuzzing function then.